tests/queries/0_stateless/02458_datediff_date32.reference

```diff
@@ -0,0 +1,169 @@
+1
+SELECT dateDiff('year', toDateTime64('1900-01-01 00:00:00', 3), toDate32('1901-01-01'));
+1
+-- With DateTime
+-- Date32 vs DateTime
+SELECT dateDiff('second', toDate32('2015-08-18'), toDateTime('2015-08-19 00:00:00'));
+86400
+SELECT dateDiff('minute', toDate32('2015-08-18'), toDateTime('2015-08-19 00:00:00'));
+1440
+SELECT dateDiff('hour', toDate32('2015-08-18'), toDateTime('2015-08-19 00:00:00'));
+24
+SELECT dateDiff('day', toDate32('2015-08-18'), toDateTime('2015-08-19 00:00:00'));
+1
+SELECT dateDiff('week', toDate32('2015-08-18'), toDateTime('2015-08-25 00:00:00'));
+1
+SELECT dateDiff('month', toDate32('2015-08-18'), toDateTime('2015-09-18 00:00:00'));
+1
+SELECT dateDiff('quarter', toDate32('2015-08-18'), toDateTime('2015-11-18 00:00:00'));
+1
+SELECT dateDiff('year', toDate32('2015-08-18'), toDateTime('2016-08-18 00:00:00'));
+1
+-- DateTime vs Date32
+SELECT dateDiff('second', toDateTime('2015-08-18 00:00:00'), toDate32('2015-08-19'));
+86400
+SELECT dateDiff('minute', toDateTime('2015-08-18 00:00:00'), toDate32('2015-08-19'));
+1440
+SELECT dateDiff('hour', toDateTime('2015-08-18 00:00:00'), toDate32('2015-08-19'));
+24
+SELECT dateDiff('day', toDateTime('2015-08-18 00:00:00'), toDate32('2015-08-19'));
+1
+SELECT dateDiff('week', toDateTime('2015-08-18 00:00:00'), toDate32('2015-08-25'));
+1
+SELECT dateDiff('month', toDateTime('2015-08-18 00:00:00'), toDate32('2015-09-18'));
+1
+SELECT dateDiff('quarter', toDateTime('2015-08-18 00:00:00'), toDate32('2015-11-18'));
+1
+SELECT dateDiff('year', toDateTime('2015-08-18 00:00:00'), toDate32('2016-08-18'));
+1
+-- With Date
+-- Date32 vs Date
+SELECT dateDiff('second', toDate32('2015-08-18'), toDate('2015-08-19'));
+86400
+SELECT dateDiff('minute', toDate32('2015-08-18'), toDate('2015-08-19'));
+1440
+SELECT dateDiff('hour', toDate32('2015-08-18'), toDate('2015-08-19'));
+24
+SELECT dateDiff('day', toDate32('2015-08-18'), toDate('2015-08-19'));
+1
+SELECT dateDiff('week', toDate32('2015-08-18'), toDate('2015-08-25'));
+1
+SELECT dateDiff('month', toDate32('2015-08-18'), toDate('2015-09-18'));
+1
+SELECT dateDiff('quarter', toDate32('2015-08-18'), toDate('2015-11-18'));
+1
+SELECT dateDiff('year', toDate32('2015-08-18'), toDate('2016-08-18'));
+1
+-- Date vs Date32
+SELECT dateDiff('second', toDate('2015-08-18'), toDate32('2015-08-19'));
+86400
+SELECT dateDiff('minute', toDate('2015-08-18'), toDate32('2015-08-19'));
+1440
+SELECT dateDiff('hour', toDate('2015-08-18'), toDate32('2015-08-19'));
+24
+SELECT dateDiff('day', toDate('2015-08-18'), toDate32('2015-08-19'));
+1
+SELECT dateDiff('week', toDate('2015-08-18'), toDate32('2015-08-25'));
+1
+SELECT dateDiff('month', toDate('2015-08-18'), toDate32('2015-09-18'));
+1
+SELECT dateDiff('quarter', toDate('2015-08-18'), toDate32('2015-11-18'));
+1
+SELECT dateDiff('year', toDate('2015-08-18'), toDate32('2016-08-18'));
+1
+-- Const vs non-const columns
```

Comment:
With the internal implementation in mind, the const vs. non-const are a bit overkill and could be omitted.

Reply:
As a base for writing 2458_datediff_date32 test I took tests/queries/0_stateless/01801_dateDiff_DateTime64.sql test which also has "const vs. non-const" blocks. BTW after finishing my 2458_datediff_date32 test and running it I found the forgotten code in "const vs. non-const" operations (I created  commit [14ba9fa](https://github.com/ClickHouse/ClickHouse/pull/42308/commits/14ba9fa51da3d505c3e040460600b07687400432) for it). So I think keeping the "const vs. non-const"  blocks of test make sense.